### ISSUE TITLE
Improve CI

### ISSUE THREAD

**Issue by jbergstroem** [arminc/clair-local-scan#19]
While filing #18 I noticed that travis seems to default to logging in and attempting to push. Perhaps we can do this on the `master` branch only? Also, I'm thinking we should build the image at travis instead of attempting to pull it?

**Comment by jbergstroem** [comment]
Perhaps we split the pipeline up into something like (pseudo-yaml only)
```yaml

build:
  script:
    - shell script to pull tag via ENV or use git sha
    - docker build ...

test:
  after: build
  script: 
    - run tests to verify both build and functionality

deploy:
  after: build
  only_when_successful
  only_on_branch: master
  script:
    - docker tag latest to tag/sha
    - docker push with tag/sha
 ```

**Comment by arminc** [comment]
Fixed with #27
